base/file.jl

```diff
@@ -436,8 +436,13 @@ function readlink(path::AbstractString)
     end
 end
 
-function chmod(p::AbstractString, mode::Integer)
-    err = ccall(:jl_fs_chmod, Int32, (Cstring, Cint), p, mode)
+function chmod(path::AbstractString, mode::Integer; recursive::Bool=false)
+    err = ccall(:jl_fs_chmod, Int32, (Cstring, Cint), path, mode)
+    if recursive && isdir(path)
```

Comment:
this should probably also do some specific checking about symbolic links, take care about whether this follows through them